### PR TITLE
Fix bleeding edge compilers

### PR DESCRIPTION
**Pull request by serban-nicusor-toptal**
Do not run the tests with --runs=0, add stanc3/test/integration/good folder to all.tests

### CODE REVIEW

**Comment by serban-nicusor-toptal** [comment]
Woohoo nice! I'll clean up the PR later and have it ready for merge & use.

**Comment by WardBrian** [comment]
@serban-nicusor-toptal do you think PRECOMPILED_HEADERS=false is required? It would probably be faster with them enabled

**Comment by serban-nicusor-toptal** [comment]
Morning! I've cleaned up the PR and removed `PRECOMPILED_HEADERS`. Ran it another time so we're everything works as expected.

File: Jenkinsfile-bleeding-edge-monthly [hunk 1]
```diff
@@ -512,15 +469,15 @@ pipeline {
...
-        always {
-            node("linux") {
-                recordIssues enabledForFailure: false, tool: clang()
-            }
-        }
-        success { script { utils.mailBuildResults("SUCCESSFUL") } }
-        unstable { script { utils.mailBuildResults("UNSTABLE") } }
-        failure { script { utils.mailBuildResults("FAILURE") } }
-    }
+    // post {
```
Comment (WardBrian) [inline Jenkinsfile-bleeding-edge-monthly:472]: We should re-enable this.

Could we also have a set list of people get emailed from the monthly run?

**Review by WardBrian — APPROVED** [review]
One thing and then good to merge! Thanks!

File: Jenkinsfile-bleeding-edge-monthly (hunk 1 above)
Comment (serban-nicusor-toptal) [reply Jenkinsfile-bleeding-edge-monthly:472]: Uncommenting this part now.
We can pass a custom list of additional emails: https://github.com/stan-dev/jenkins-shared-libraries/blob/master/src/org/stan/Utils.groovy#L236

**Pull request merged**